lib/HALInterfaces/src/IDevice.h

```diff
@@ -101,7 +101,7 @@ class IDevice
      *
      * @return If device is in bootloader mode, it will return true. Otherwise false.
      */
-    virtual bool isInBootloaderMode() const = 0;
+    virtual bool isInBootloaderMode() = 0;

```

Comment:
Removed because m_acm.isReady() misses the const?

Reply:
Yes! Is there a better way to do this?

Reply:
You can add
```mutual XXXXX m_acm;```
Fast solution, but not nice.

Reply:
I am not sure what the mutual does.
Should I still make the change?

Reply:
mutable not mutual ?

It removes "const" qualifier from the element it is used at.

Reply:
I would try this, its bad if an internal used class enforces to remove const.